### PR TITLE
Use arraytxt for object bonus description

### LINKED ISSUE #962: buggy morale strings
1. Sepulcher (Crypt) has extra line break
2. Crypt and Warrior Tomb strings contain numbers twice

Screenshot 2022-09-21 at 12 35 25

Also, in oh3 Warrior Tomb has different message:
Screenshot 2022-09-21 at 15 54 05

### PR DESCRIPTION
**Pull request by gamestales**
Fixes #962

### CODE REVIEW

**Comment by gamestales** [comment]
screen
So it works but it sets map object description on the fly.

It would be best to fill it during initialization.
Also probably should pass bonus type and object type instead of description.

File: lib/mapObjects/CRewardableObject.cpp [hunk 1]
```diff
@@ -83,8 +83,21 @@ void CRewardableObject::onHeroVisit(const CGHeroInstance *h) const
...
+			for(auto index : rewards)
+			{
+				auto& reword = configuration.info[index].reward;
+				for(auto& bonus : reword.bonuses)
+				{
+					if(bonus.type == BonusType::MORALE && bonus.description.empty())
+						const_cast<std::string&>(bonus.description) = VLC->generaltexth->arraytxt[104]; "\nWarrior Tomb Visited -3";
+				}
+			}
+		}
```
Comment (gamestales) [inline lib/mapObjects/CRewardableObject.cpp:99]: This setup part could be filled somewhere else. I have no idea where.

File: lib/networkPacks/NetPacksLib.cpp [hunk 2]
```diff
@@ -993,25 +993,21 @@ void GiveBonus::applyGs(CGameState *gs)
...
-	if(bdescr.empty() && (bonus.type == BonusType::LUCK || bonus.type == BonusType::MORALE))
+	if(!bdescr.empty()) 
 	{
-		if (bonus.source == BonusSource::OBJECT_TYPE || bonus.source == BonusSource::OBJECT_INSTANCE)
-		{
-			descr = VLC->generaltexth->arraytxt[bonus.val > 0 ? 110 : 109]; //+/-%d Temporary until next battle"
-		}
-		else if(bonus.source == BonusSource::TOWN_STRUCTURE)
-		{
-			descr = bonus.description;
```
Comment (gamestales) [inline lib/networkPacks/NetPacksLib.cpp:1004]: This is actully copied before.

File: lib/networkPacks/NetPacksLib.cpp [hunk 3]
```diff
@@ -993,25 +993,21 @@ void GiveBonus::applyGs(CGameState *gs)
...
-			return;
-		}
-		else
-		{
-			descr = bdescr.toString();
```
Comment (gamestales) [inline lib/networkPacks/NetPacksLib.cpp:1009]: bdescr is empty here

File: lib/mapObjects/CRewardableObject.cpp (hunk 1 above)
Comment (IvanSavenko) [reply lib/mapObjects/CRewardableObject.cpp:100]: Should be in object config, in json. Not hardcoded

File: lib/networkPacks/NetPacksLib.cpp [hunk 4]
```diff
@@ -993,25 +993,21 @@ void GiveBonus::applyGs(CGameState *gs)
...
-		}
+		descr = bdescr.toString();
+	}
+	else if(descr.size()) {} //use preseet description
```
Comment (IvanSavenko) [inline lib/networkPacks/NetPacksLib.cpp:1000]:
```suggestion
	else if(!descr.empty()) 
	{
		//use preset description
	} 
```

**Pull request merged**